File: backend/src/main/java/com/phonebook/backend/classes/Group.java
package com.phonebook.backend.classes;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.*;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "groups")
@Getter
@Setter
@NoArgsConstructor
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Group {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    @JsonIgnoreProperties(value = "groups")
    @ManyToMany(mappedBy = "groups")
    private List<Person> persons = new ArrayList<>();

    public Group(String name) {
        this.name = name;
    }

}

File: front/src/components/Group.js
import React from 'react'

const Group = ({ group, deleteGroup }) => {
  return (
    <div>
      ({group.id}) <b>Group name</b>: {group.name} <button id={group.id} name={group.name} onClick={deleteGroup}>delete group?</button>
      {group.persons.map(person => <li key={person.id}>{person.name}</li>)}
    </div>
  )
}

export default Group